File: ch5/Queue.py
class Queue:
    def __init__(self):
        self.queue = []

    def insert(self,value):
        self.queue.append(value)

    def dequeue(self):
        if len(self.queue) == 0:
            return None
        first_elt = self.queue[0]
        del self.queue[0]
        return first_elt

    def empty(self):
        return self.queue == []
}

File: cpp/Ch5/Queue.cpp
#include <vector>
#include <iostream>

class Queue {
private:
    std::vector<int> queue;

public:
    void insert(int value) {
        queue.push_back(value);
    }

    int dequeue() {
        if (queue.empty()) {
            return -1; 
        }
        int first_elt = queue.front();
        queue.erase(queue.begin()); 
        return first_elt;
    }

    bool empty() const {
        return queue.empty();
    }
};